internal/clients/telegram/state/daily_status.go

```diff
@@ -0,0 +1,235 @@
+			s.DiscoveryOfTheDay = option.None[string]()
+		} else {
+			s.DiscoveryOfTheDay = option.Some(text)
+		}
+
+		return NewTransition(s.DailyStatusState, s.userData, []response.BotAction{
+			response.NewSendMessage(response.ChatID(fmt.Sprint(chatID)),
+				s.responses.QuestionsAndBlockers,
+			),
+		})
+
+	case questionsAndBlockersDailyStatusStage:
+		if isCmd && strings.ToLower(cmd.Method) == noneCommand {
+			s.QuestionsAndBlockers = option.None[string]()
+		} else {
+			s.QuestionsAndBlockers = option.Some(text)
+		}
+
+		report, err := s.generateReport()
+		if err != nil {
+			report = github.GqlErrorStringOr("Github API error: %s", err, "Something went wrong while contacting GitHub.")
+		}
+
+		return NewTransition(s.RootState, s.userData, []response.BotAction{
+			response.NewSendMessage(response.ChatID(fmt.Sprint(chatID)), report),
+		})
+	}
+
+	return s.Ignore()
+}
+
+/*
+Checks if the query is to save the default project. The retuned bool indicates if the query was about saving the default
+project. If returns false try other queries if present or return an ignore response.
+*/
+func (s *DailyStatusHandler) handleCQSaveDefaultProject(data string, message update.Message) (Transition, bool) {
+	askDiscoveryOfTheDay := []response.BotAction{
+		response.RemoveReplyMarkup(message),
+		response.NewSendMessage(response.ChatID(fmt.Sprint(message.Chat.ID)), s.responses.DiscoveryOfTheDay),
+	}
+
+	if data == cqDailyStatusAskDefaultProjectEveryTime {
+		s.UseOnlyProjectNoSaveDefault = true
+
+		return NewTransition(s.DailyStatusState, s.userData, askDiscoveryOfTheDay), true
+	}
+
+	if data == cqDailyStatusSetOnlyProjectAsDefault {
+		s.DefaultProject = option.Some[github.ProjectID](s.UseProject)
+
+		return NewTransition(s.DailyStatusState, s.userData, askDiscoveryOfTheDay), true
+	}
+
+	return Transition{}, false
+}
+
+func (s *DailyStatusHandler) generateReport() (string, error) {
+	items, err := github.NewClient(s.userData.GithubAPIKey.UnwrapOr("")).ListViewerProjectV2Items(context.Background(),
+		s.UseProject, option.Some(dailyStatusItemLimit), option.None[github.ProjectCursor]())
+	if err != nil {
+		return "", errors.WithMessage(err, "while getting user's project v2 items")
+	}
+
+	const listSep = "\n• "
+
+	report := fmt.Sprintf(`#daily report %s:
+<b>Today I worked on</b>%s
+
+<b>Tomorrow I will work on</b>%s
+
+`, time.Now().Format("01.02"), listSep+strings.Join(items["Done"], listSep),
+		listSep+strings.Join(items["In Progress"], listSep))
+
+	if dod, isSome := s.DiscoveryOfTheDay.Unwrap(); isSome {
+		report += "<b>Discovery of the day</b>\n\n" + dod + "\n\n"
+	}
+
+	if blockers, isSome := s.QuestionsAndBlockers.Unwrap(); isSome {
+		report += "<b>Questions/Blockers</b>\n\n" + blockers + "\n\n"
+	}
+
+	if len(items["In Review"]) != 0 {
+		report += "<b>In review</b>" + listSep + strings.Join(items["In Review"], listSep)
+	}
+
+	return report, nil
+}
+
+type DailyStatusState struct {
```

Comment:
Add an optional date override to this? It would come from `/dailyStatus 01.02` and would override the default date that is from time.Now().